src/item_contents.cpp

```diff
@@ -495,6 +495,9 @@ ret_val<item_pocket *> item_contents::insert_item( const item &it,
     }
 
     ret_val<item_pocket *> pocket = find_pocket_for( it, pk_type );
+    if( !pocket.value()->is_allowed() ) {
+        return ret_val<item_pocket *>::make_failure( nullptr, _( "Can't store anything in this." ) );
+    }
     if( !pocket.success() ) {
         return pocket;
     }
```

Comment:
Seems to be triggering a segfault in the tests since the pointer is invalid.
```suggestion
    if( !pocket.success() ) {
        return pocket;
    }
    if( !pocket.value()->is_allowed() ) {
        return ret_val<item_pocket *>::make_failure( nullptr, _( "Can't store anything in this." ) );
    }
```